packages/web-workers/src/controllers/scan-batch-request-feed-controller.ts

```diff
@@ -140,33 +138,40 @@ export class ScanBatchRequestFeedController extends WebController {
     }
 
     private createWebsiteScanResult(request: ScanRunBatchRequest): WebsiteScanResult {
+        let consolidatedGroup;
         if (request.reportGroups !== undefined) {
-            const consolidatedGroup = request.reportGroups.find((group) => group.consolidatedId !== undefined);
-            if (consolidatedGroup) {
-                const websiteScanRequest: Partial<WebsiteScanResult> = {
-                    baseUrl: request.site?.baseUrl,
-                    scanGroupId: consolidatedGroup.consolidatedId,
-                    // `deepScanId` value is set only when db document is created
-                    deepScanId: request.deepScan ? request.scanId : undefined,
-                    scanGroupType: request.deepScan ? 'deep-scan' : 'consolidated-scan-report',
-                    pageScans: [
-                        {
-                            scanId: request.scanId,
-                            url: request.url,
-                            timestamp: new Date().toJSON(),
-                        },
-                    ],
-                    knownPages: request.site?.knownPages,
-                    discoveryPatterns: request.site?.discoveryPatterns?.length > 0 ? request.site.discoveryPatterns : undefined,
-                    // `created` value is set only when db document is created
-                    created: new Date().toJSON(),
-                };
+            consolidatedGroup = request.reportGroups.find((g) => g.consolidatedId !== undefined);
+        }
 
-                return this.websiteScanResultProvider.normalizeToDbDocument(websiteScanRequest);
-            }
+        let scanGroupType: ScanGroupType;
+        if (request.deepScan === true) {
+            scanGroupType = 'deep-scan';
+        } else if (consolidatedGroup?.consolidatedId || request.site?.knownPages?.length > 0) {
+            scanGroupType = 'consolidated-scan';
+        } else {
+            scanGroupType = 'single-scan';
```

Comment:
How will this change impact privacy scans which currently specify deepScan true and pass knownPages?

Reply:
Will not be affected. The `deepScan` will be optional.